src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,96 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_SIZE = 10;
+    private int lastIndexOfArray = 0;
+    private int index = 0;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ARRAY_SIZE];
+        values = (V[]) new Object[MAX_ARRAY_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key != null) {
+            if (lastIndexOfArray == 0) {
+                saveStorage(key, value);
+            } else if (isKey(key)) {
+                values[index] = value;
+            } else {
+                saveStorage(key, value);
+            }
+            index = 0;
+        } else if (isKey(key)) {
+            values[index] = value;
+        } else {
+            saveStorage(null, value);
+        }
     }
 
     @Override
     public V get(K key) {
+        index = 0;
+        for (; index < lastIndexOfArray; index++) {
+            if (isKey(key)) {
+                return values[index];
+            }
+        }
         return null;
```

Comment:
```suggestion
        for (int i = 0; i < size; i++) {
            if (keysEquals(key, keys[i])) {
                return values[index];
            }
        }
        return null;
```
index should be local - here index is the same as i